src/domain/usecases/data-entry/amc/RecalculateConsumptionDataProductLevelForAllUseCase.ts

```diff
@@ -0,0 +1,303 @@
+import {
```

Comment:
Without knowing the intricacies or history of the code, it seems that data repositories are huge, containing a lot of logic. Not saying we have to change this, but typically repositories should be logic-less, just get/store the data. The logic is in use cases and entities. Note more for the future than for that specific PR.